### ISSUE TITLE
Static code shouldn't be generated

### ISSUE THREAD

**Issue by MisinformedDNA** [reactivemarbles/ObservableEvents#136]
The string in [EventGenerator.cs](https://github.com/reactivemarbles/ObservableEvents/blob/main/src/ReactiveMarbles.ObservableEvents.SourceGenerator/EventGenerator.cs#L30-L72) is never altered, so why is this code being generated. Why not just create actual classes so they don't have to be generated?

**Comment by glennawatson** [comment]
Without it the entire generator wouldn't work.

It uses the code to be injected in as a 'hook' then relies on calls to the
generic class in the class to produce the actual observable.

**Comment by MisinformedDNA** [comment]
I [tried it](https://github.com/MisinformedDNA/ObservableEvents/tree/less-generation) and I didn't notice any problems. If you look at the .NET code base, their attributes aren't generated either. See https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/Common/JsonSourceGenerationOptionsAttribute.cs and https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LoggerMessageAttribute.cs.

It's especially important for the attribute to be accessible for discoverability purposes. As it is now, if I paste `[assembly: GenerateStaticEventObservables(typeof(StaticTest))]` into my code, VS can't help me identify what `using` statement I'm missing. It's a small thing that helps developers fall into the pit of success.

**Comment by glennawatson** [comment]
Yeah, it's changing slightly how it is working in the next release but not from being injected by the source generator.

Mostly the next version which is Roslyn 4-based will avoid the use of namespaces.

System.text.Json and the Microsoft logging extensions have the luxury of an existing assembly.

**Comment by glennawatson** [comment]
The approach you've taken in your example won't work as a package fyi, due to it being a development dependency library and a analyzer.

Thanks for bringing up the suggestion regardless.

**Comment by MisinformedDNA** [comment]
Yeah, .NET does have that luxury. Follow-up question: Why does it have to be a development dependency? I searched NuGet. Some are and some aren't.

**Comment by glennawatson** [comment]
If you don't have it as a development dependency then every project that depends on this source generator will have the source generator running. We opt to have it for the user to opt in each time for each project by default.
